Source/EmbeddedBoundary/WarpXFaceExtensions.cpp

```diff
@@ -0,0 +1,1114 @@
+                    // multi-face extensions.
+                    if (S_red(i, j + j_n, k + k_n) > S_ext
+                        and (flag_info_face(i, j + j_n, k + k_n) == 1 or
+                             flag_info_face(i, j + j_n, k + k_n) == 2)
+                        and flag_ext_face(i, j, k) and not stop) {
+                        n_borrow += 1;
+                        stop = true;
+                    }
+                }
+            }
+        }
+    }else if(idim == 1){
+        for (int i_n = -1; i_n < 2; i_n++) {
+            for (int k_n = -1; k_n < 2; k_n++) {
+                //This if makes sure that we don't visit the "diagonal neighbours"
+                if (!(i_n == k_n or i_n == -k_n)) {
+                    // Here a face is available if it doesn't need to be extended itself and if its
+                    // area exceeds Sz_ext. Here we need to take into account if the intruded face
+                    // has given away already some area, so we use Sz_red rather than Sz.
+                    // If no face is available we don't do anything and we will need to use the
+                    // multi-face extensions.
+                    if (S_red(i + i_n, j, k + k_n) > S_ext
+                        and (flag_info_face(i + i_n, j, k + k_n) == 1
+                             or flag_info_face(i + i_n, j, k + k_n) == 2)
+                        and flag_ext_face(i, j, k) and not stop) {
+                        n_borrow += 1;
+                        stop = true;
+                    }
+                }
+            }
+        }
+    }else if(idim == 2) {
+        for (int i_n = -1; i_n < 2; i_n++) {
+            for (int j_n = -1; j_n < 2; j_n++) {
+                //This if makes sure that we don't visit the "diagonal neighbours"
+                if (!(i_n == j_n or i_n == -j_n)) {
+                    // Here a face is available if it doesn't need to be extended itself and if its
+                    // area exceeds Sz_ext. Here we need to take into account if the intruded face
+                    // has given away already some area, so we use Sz_red rather than Sz.
+                    // If no face is available we don't do anything and we will need to use the
+                    // multi-face extensions.
+                    if (S_red(i + i_n, j + j_n, k) > S_ext
+                        and (flag_info_face(i + i_n, j + j_n, k) == 1
+                             or flag_info_face(i + i_n, j + j_n, k) == 2)
+                        and flag_ext_face(i, j, k) and not stop) {
+                        n_borrow += 1;
+                        stop = true;
+                    }
+                }
+            }
+        }
+    }
+
+    return n_borrow;
+}
+
+/**
+* \brief For the face of cell pointing in direction idim, compute the number of faces
+* we need to intrude. For the one-way extension this function returns only one or zero: one if the
+* face can be extended withe the one-way extension, zeros if it can't.
```

Comment:
Was it like this (or was it "within the")?
```suggestion
* face can be extended with the the one-way extension, zeros if it can't.
```